Consolidate time series downsampling docs

marciw opened the pull request:
This PR restructures and partially revises the downsampling section:
* remove repetitive info and filler
* apply a more logical structure 
* consolidate methods on one page with tabs
* edit for clarity, conciseness, and Elastic style (partial)

closes https://github.com/elastic/docs-content/issues/2239

TODO: Create issue for "fast follow" improvements, in the interest of time

yannis-roussos commented on manage-data/data-store/data-streams/downsampling-concepts.md:
```diff
@@ -0,0 +1,99 @@
+
+The downsampling operation traverses the source TSDS index and performs the following steps:
+
+1. Creates a new document for each value of the `_tsid` field and each `@timestamp` value, rounded to the `fixed_interval` defined in the downsampling configuration.
```
I know that this was copied from the previous concepts page, but I still find it very confusing.

What we want to say here is that we gather all the documents that have the same dimension values (that's what the tsid is in reality) and timestamps inside a specific interval (we call this a bucket, which is also referenced later out of the blue). And we then create a single document for all of them with the same values for the dimensions (tsid) and as its timestamp the start of the interval

e.g. I have:
```
{time="8:15:21", location="paris", quartier ="belleville", temperature=22.1, wind=1.1}
{time="8:15:23", location="paris", quartier ="belleville", temperature=22.5, wind=1.2}
{time="8:15:24", location="paris", quartier ="belleville", temperature=21.9, wind=1.1}
{time="8:15:27", location="paris", quartier ="belleville", temperature=22.2, wind=1.3}
```

If we downsample per 10 seconds, those 4 documents would be summarized to one:
```
{time="8:15:20", location="paris", quartier ="belleville", STATS for the temp and wind on those 4 docs}
```

So, I think that this could be better, but be my guest to change it further
```suggestion
1. Creates a new document for each group of documents with the same time series dimension values (same `_tsid` value) and downsampling bucket (timestamps that fall in a specific interval)
```

We could even skip the parentheses at this level

yannis-roussos reviewed (commented):
Thank you @marciw, this revamp looks really great! I have added some small comments and pinged Mary to help with an additional review / pair of eyes - her expertise on downsampling will be invaluable to iron out the last few details and move forward!

gmarouli commented on manage-data/data-store/data-streams/run-downsampling.md:
```diff
@@ -0,0 +1,157 @@
+
+:::{tab-item} Data stream lifecycle
+
+## Downsampling with data stream lifecycle management
```
Unless it changed recently the feature is called `data stream lifecycle` not `data stream lifecycle management` or `data lifecycle management`. Leaving this as a note to ensure the docs mentions this properly.

gmarouli commented on manage-data/data-store/data-streams/run-downsampling.md:
````diff
@@ -0,0 +1,157 @@
+```
+
+
+For more details about index templates for time series data streams, refer to [](set-up-tsds.md).
````
This should be applicable for both ILM & data stream lifecycle, right?

marciw replied:
Moved this down, thanks

gmarouli commented on manage-data/data-store/data-streams/run-downsampling.md:
```diff
@@ -0,0 +1,157 @@
+
+## Downsampling with data stream lifecycle management
+
+To downsample time series data as part of data lifecycle management, create an index template that includes a `lifecycle` section with a [downsampling](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-data-lifecycle) object. 
```
This sentence is confusing different concepts:

-  data lifecycle management refers to both ILM & data stream lifecycle
- It suggests to add the lifecycle section in an index template but the docs it refers to is of the put lifecycle API which has nothing to do with the index template. 

In ILM we suggest the user to add downsampling to their current policy, should we suggest the same here? Add downsampling to the lifecycle, and assume the user knows what we are talking about?

```suggestion
To downsample a TSDS using data stream lifecycle, you can add a downsampling section in the lifecycle. 
```

gmarouli commented on manage-data/data-store/data-streams/run-downsampling.md:
```diff
@@ -0,0 +1,157 @@
+To downsample time series data as part of data lifecycle management, create an index template that includes a `lifecycle` section with a [downsampling](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-data-lifecycle) object. 
+
+* Set `fixed_interval` to your preferred level of granularity. The original time series data will be aggregated at this interval.
+* Set `after` to the minimum time to wait after an index rollover, before running downsampling.
```
```suggestion
* Set `after` to the minimum time to wait after a backing index rolled over, before running downsampling.
```

marciw replied:
Didn't make this change

gmarouli commented on manage-data/data-store/data-streams/run-downsampling.md:
````diff
@@ -0,0 +1,157 @@
+* Set `after` to the minimum time to wait after an index rollover, before running downsampling.
+
+```console
+PUT _index_template/datastream_template
+{
+  "index_patterns": [
+    "datastream*"
+  ],
+  "data_stream": {},
+  "template": {
+    "lifecycle": {
+      "downsampling": [
+        {
+          "after": "1m",
+          "fixed_interval": "1h"
+        }
+      ]
+    },
+    "settings": {
+      "index": {
+        "mode": "time_series"
+      }
+    },
+    "mappings": {
+      "properties": {
+        "@timestamp": {
+          "type": "date"
+        },
+        [...]
+      }
+    }
+  }
+}
````
If we are talking about existing TSDS the following could be enough assuming the data stream is already managed by data stream lifecycle 
```suggestion
PUT _data_stream/my-data-stream/_lifecycle
{
  "data_retention": "7d",
  "downsampling": [
     {
       "after": "1m",
       "fixed_interval": "10m"
      },
      {
        "after": "1d",
        "fixed_interval": "1h"
      }
   ]
}
```
What do you think?

marciw replied:
Made this change, thanks!

marciw commented on manage-data/data-store/data-streams/downsampling-time-series-data-stream.md:
```diff
@@ -8,145 +9,28 @@ products:
-:alt: time series original
-:title: Original metrics series
-:::
+Metrics tools and solutions collect large amounts of time series data over time. As the data ages, it becomes less relevant to the current state of the system. _Downsampling_ lets you reduce the resolution and precision of older data, in exchange for increased storage space.
```
@yannis-roussos I'm going to revert this change -- it was syntactically/idiomatically correct as it was. You downsample, and the benefit you get "in exchange for downsampling" is increased storage space. It's a trade-off.

I think the "in exchange for" structure is idiomatic and concise, but we could spell it out more if you prefer:

"_Downsampling_ lets you reduce the resolution and precision of older data, in order to free up storage space."
or
"_Downsampling_ lets you reduce the resolution and precision of older data, in order to also reduce the data's storage footprint."

The last option seems good. WDYT?

marciw commented on manage-data/data-store/data-streams/downsampling-time-series-data-stream.md:
```diff
@@ -8,145 +9,28 @@ products:
-:alt: time series downsampled
-:title: Downsampled metrics series
+:::{tip}
+You can include downsampling in an [{{ilm}} ({{ilm-init}})](../../lifecycle/index-lifecycle-management.md) policy to automatically manage the volume and associated cost of your metrics data as it ages.
```
added applies_to tag for the section

marciw commented on manage-data/data-store/data-streams/run-downsampling.md:
```diff
@@ -0,0 +1,157 @@
+
+Before you start, make sure your index is a candidate for downsampling:
+
+* The index must be **read-only**. You can roll over a write index and make it read-only.
```
I tried to address some of this, but please take another look -- not sure I'm following your comments.

marciw commented:
@yannis @mary I've addressed most of your comments and asked for clarification on a few others. Please take another look when you have time. 🙏 

Also note that this is just one piece of the overall restructuring -- additional PRs coming soon

yannis-roussos commented on manage-data/data-store/data-streams/downsampling-concepts.md:
```diff
@@ -0,0 +1,99 @@
+4. For all other fields, copies the most recent value to the target index.
+5. Deletes the original index and replaces it with the downsampled index. Within a data stream, only one index can exist for a time period.
+
+The new, downsampled index is created on the data tier of the original index and inherits the original settings, like number of shards and replicas.
```
Makes sense Mary, let's keep it as is then!

yannis-roussos commented on manage-data/data-store/data-streams/downsampling-concepts.md:
```diff
@@ -0,0 +1,99 @@
+* You can query multiple raw data and downsampled indices in a single request, and a single request can include downsampled indices with multiple downsampling intervals (for example, `15m`, `1h`, `1d`).
+* When you run queries in {{kib}} and through Elastic solutions, a standard response is returned, with no indication that some of the queried indices are downsampled.
+* [Date histogram aggregations](elasticsearch://reference/aggregations/search-aggregations-bucket-datehistogram-aggregation.md) support `fixed_intervals` only (not calendar-aware intervals).
+* Time-based histogram aggregations use a uniform bucket size, without regard to the downsampling time interval specified in the request.
```
I agree, the new dedicated page for querying downsampled data looks much simple now. Maybe just add a note to link to the TS command once it is ready (as an update included in https://github.com/elastic/elasticsearch/pull/134373)?

marciw commented on manage-data/data-store/data-streams/run-downsampling.md:
````diff
@@ -0,0 +1,124 @@
+serverless: ga
+```
+
+To downsample a time series via a data stream lifecycle, add a [downsampling](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-data-lifecycle) section to the lifecycle. 
````
Fixed in d5af0d309ff414c26eca7260bb52f5cd04aa1211

leemthompo reviewed (approved):
Looks like SMEs are happy, changes LGTM 👍.

I just wonder if we should pay attention to the H1s and add additional context 🤷.

For example, should we have stuff like :  

```diff
- Downsampling concepts
+ How downsampling in {{es}} works
```

marciw commented:
> I just wonder if we should pay attention to the H1s and add additional context 🤷.

🙏 yes! coming soon in a follow-up PR (I did these in phases and I think that ended up being confusing)
